bin/cli.js

```diff
@@ -1,17 +1,171 @@
+  'PhantomJS': 'karma-phantomjs-launcher',
+  'Safari': 'karma-safari-launcher'
+};
+
+const rmDuplicates = arr => arr.reduce((acc, cur) => {
+  if (acc.indexOf(cur) < 0) acc.push(cur);
+  return acc;
+}, []);
+
+const loadFiles = (files) => {
+  const result = [];
+  files.forEach(file => {
+    if (fs.statSync(file).isDirectory()) {
+      const subfiles = fs.readdirSync(file).map(f => `${file}/${f}`);
+      result.push(...loadFiles(subfiles));
+    } else if (file.indexOf('.js') > 0) {
+      result.push(file);
+    }
+  });
+  return result;
+};
+
+const setKarmaBrowsers = (config, ...browsers) => {
+  browsers = rmDuplicates(browsers);
 
-walk('../test', (err, files) => {
-  metasync.for(files).each((file, next) => {
-    require(file.filePath);
-    next();
-  }).fetch(() => {
-    console.log('Tests done');
+  if (!config.plugins) config.plugins = [];
+  if (!config.browsers) config.browsers = [];
+  browsers.forEach(browser => {
+    const launcher = browserLaunchers[browser];
+    if (launcher && !config.browsers[browser]) {
+      config.browsers.push(browser);
+      config.plugins.push(launcher);
+    }
   });
-});
+  return config;
+};
+
+const setKarmaLogLevel = (config, logLevel = 'disable') => {
+  if (!config.webpackMiddleware) config.webpackMiddleware = {};
+  config.logLevel = karma.constants[karmaLogLevels[logLevel][0]];
+  config.webpackMiddleware.stats = karmaLogLevels[logLevel][1];
+  return config;
+};
+
+const getPreprocessor = (files) => {
+  const createPreprocessor = () => (content, file, done) =>
+    done(`metafiles=${JSON.stringify(files)};${content}`);
+  createPreprocessor.$inject = [];
+  return createPreprocessor;
+};
+
+const getBrowserConfig = (conf) => {
+  const config = {
+    plugins: [
+      { 'preprocessor:metatests': ['factory', getPreprocessor(conf.files)] },
+      'karma-webpack'
+    ],
+    webpackMiddleware: {},
+    webpack: { node: {} },
+    preprocessors: {},
+    files: [],
+    basePath: process.env.PWD,
+    port: conf.browser.port,
+    autoWatch: false,
+    singleRun: true,
+    concurrency: 1
+  };
+
+  const adapter = path.resolve(__dirname, '../karma/adapter.js');
+  config.files.push(adapter);
+  config.preprocessors[adapter] = ['webpack', 'metatests'];
+
+  [ 'fs', 'child_process' ].forEach(lib => config.webpack.node[lib] = 'empty');
+  setKarmaBrowsers(config, ...conf.browser.browsers);
+  setKarmaLogLevel(config, conf.browser.logLevel);
+  return config;
+};
+
+const getConfig = () => {
+  const version = parseFile(path.resolve(__dirname, '../package.json')).version;
+  program.version(version).usage('[options] -- <file ...>');
+  clOptions.forEach(option => program.option(...option));
+  program.parse(process.argv);
+
+  const config = program.config ? parseFile(program.config) : {};
+
+  config.enviroment = program.enviroment;
+  config.files = config.files || [];
+  config.files.push(...program.args);
+  config.files = config.files.map(file => `test/${file}`);
```

Comment:
This is ok, but it is impossible for browser tests, because `webpack` have to parse files before loading modules. This is possible:
```js
require('./test/' + filename);
```
And this is not:
```js
const dir = './test/';
require(dir + filename);
```